pkg/csi/service/service.go

```diff
@@ -106,22 +106,47 @@ func (s *service) BeforeServe(
 		log.Infof("configured: %q with clusterFlavor: %q and mode: %q", csitypes.Name, clusterFlavor, s.mode)
 	}()
 
+	var (
+		err error
+		cfg *cnsconfig.Config
+	)
+
+	cfg, err = common.GetConfig(ctx)
+	if err != nil {
+		log.Errorf("failed to read config. Error: %+v", err)
+		return err
+	}
+
 	// Get the SP's operating mode.
 	s.mode = csictx.Getenv(ctx, gocsi.EnvVarMode)
 	if !strings.EqualFold(s.mode, "node") {
 		// Controller service is needed
-		var cfg *cnsconfig.Config
-		var err error
-
-		cfg, err = common.GetConfig(ctx)
-		if err != nil {
-			log.Errorf("failed to read config. Error: %+v", err)
-			return err
-		}
 		if err := s.cnscs.Init(cfg); err != nil {
 			log.Errorf("failed to init controller. Error: %+v", err)
 			return err
 		}
+	} else {
+		// Initialize CO utility in Nodes
+		var k8sInitParams interface{}
+		if clusterFlavor == cnstypes.CnsClusterFlavorVanilla {
+			k8sInitParams = k8sorchestrator.K8sVanillaInitParams{
+				InternalFeatureStatesConfigInfo: cfg.InternalFeatureStatesConfig,
+			}
+		} else if clusterFlavor == cnstypes.CnsClusterFlavorGuest {
+			k8sInitParams = k8sorchestrator.K8sGuestInitParams{
+				InternalFeatureStatesConfigInfo:   cfg.InternalFeatureStatesConfig,
+				SupervisorFeatureStatesConfigInfo: cfg.FeatureStatesConfig,
+			}
+		} else {
+			mssg := fmt.Sprintf("unrecognized mode %q in cluster flavor %q", s.mode, clusterFlavor)
+			log.Error(mssg)
+			return errors.New(mssg)
+		}
+		containerOrchestratorUtility, err = commonco.GetContainerOrchestratorInterface(ctx, common.Kubernetes, clusterFlavor, k8sInitParams)
```

Comment:
Where is `containerOrchestratorUtility` defined?

Reply:
https://github.com/kubernetes-sigs/vsphere-csi-driver/pull/362/files#diff-bb7a60c0070529376294772c178047ba865648b273d5f7d4f58742c12cdb7f67R63. It is defined in `node.go`